### PR TITLE
feat(wasm): support wasm call host http API

### PR DESCRIPTION
**Pull request by huangzhiran**
https://iotex.larksuite.com/docx/IAODdYkwFowx03x8HBtu8YUcsKb

### CODE REVIEW

File: http/http.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package http
```
Comment (Liuhaai) [inline http/http.go]: the package name `http` is bad which will cause a collision. Mb `api.Call()` insead of `http.Do()`?

File: http/http.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"bytes"
+	"errors"
+	"net/http"
+	"unsafe"
+
+	"github.com/machinefi/w3bstream-wasm-golang-sdk/common"
+)
+
+func Do(req *http.Request) (*http.Response, error) {
+	req.URL.Host = "w3bstream.com"
```
Comment (Liuhaai) [inline http/http.go]: how is it used in host side?

File: http/http.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	req.URL.Scheme = "w3bstream"
```
Comment (Liuhaai) [inline http/http.go]: shoud `host` and `scheme` be overwritten in SDK?

File: http/http.go (hunk 2 above)
Comment (huangzhiran) [reply http/http.go]: go http package need the Host for routing.

File: examples/hello/main.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+func main() {}
+
+//export start
+func _start(rid uint32) int32 {
+	req, err := http.NewRequest("GET", "/system/hello", nil)
+	if err != nil {
+		return -1
+	}
+	req.Header.Set("name", "w3bstream")
```
Comment (Liuhaai) [inline examples/hello/main.go:19]: will it work without this line?

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**